### PR TITLE
Fix/dtrsd 19625

### PR DESCRIPTION
**Pull request by JeremyDec**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | yes
| Tickets          | Fix # DTRSD-19625
| License          | BSD 3-Clause

## Description

Handle new pricings with 'create-default' for pricing mode.

BREAKING CHANGE: 6b12f74
(Adaptation is made by 8fe6a85 & cba976b)

### CODE REVIEW

File: packages/manager/apps/web/client/app/domain/zone/activate/activate.controller.js [hunk 1]
```diff
@@ -12,20 +13,29 @@ export default class DomainDnsZoneActivateController {
...
-      this.serviceOption.prices,
-      (price) => price.pricingMode === pricingConstants.PRICING_MODE.DEFAULT,
+    this.prices = uniqBy(
+      filter(
+        this.serviceOption.prices,
+        (price) =>
+          [
+            pricingConstants.PRICING_MODE.DEFAULTS.CREATE,
+            pricingConstants.PRICING_MODE.DEFAULTS.DEFAULT,
+          ].includes(price.pricingMode) && price.interval !== 0,
```
Comment (JeremyDec) [inline packages/manager/apps/web/client/app/domain/zone/activate/activate.controller.js:36]: Ugly solution, we should be able to not filter many thing to have correct pricings, but this requires a drastic change. (Use of other routes)

File: packages/manager/apps/web/client/app/domain/zone/activate/activate.controller.js [hunk 2]
```diff
@@ -12,20 +13,29 @@ export default class DomainDnsZoneActivateController {
...
+      ),
+      'interval',
```
Comment (JeremyDec) [inline packages/manager/apps/web/client/app/domain/zone/activate/activate.controller.js:38]: Ugly solution, we should be able have correct pricings without making all this stuff, but this requires a bigger change in the worflow. (Use of other routes etc..)

**Comment by marie-j** [comment]
@JeremyDec as you introduce a breaking change, some semvers will not be compatible anymore

**Comment by JeremyDec** [comment]
@marie-j yes, I'll adapt the apps using the package

**Review by marie-j — APPROVED** [review]

**Pull request merged**